src/main/java/frc/robot/Plunger.java

```diff
@@ -2,29 +2,224 @@
 
 //imports
 import edu.wpi.first.wpilibj.Solenoid;
+import edu.wpi.first.wpilibj.AnalogInput;
+import edu.wpi.first.wpilibj.Joystick;
+import edu.wpi.first.wpilibj.Timer;
 
 /**
  *  Controls plunger mechanism.
  */
 public class Plunger{
 
     // object declaration
-    Solenoid solenoid;
+    Solenoid upstreamSolenoid;
+    Solenoid downstreamSolenoid;
+    AnalogInput pressureSensor;
+    AnalogInput vacuumSensor;
+    plungerState state;
+    Timer timer;
+    Joystick joy;
```

Comment:
Don't use Joysticks outside the robot class! Please remove this and associated commands.